### PR TITLE
Merge new data module and reorganize data files

### LINKED ISSUE #38: Shift data files into data-generating scripts
### Task
Currently `src/menelaus` contains `tools/artifacts` which only exist to house large CSV files for test datasets, and accompanying descriptions. As many of these as possible should be transformed such that the data does not need to live in-repository, and can instead be generated by Python scripts instead. For now, this applies only to `example_data.csv` via `make_example_data.R`.

### Impact
Transforming CSVs into generated data via Python scripts allows for greater flexibility for users, and mimics patterns in `tensorflow.keras.datasets`, `sklearn.datasets`, etc. This will also allow us to refactor the suboptimal "tools" folder, which isn't really a sub-package at the moment and contains some large files it may be preferable to avoid downloading. 

### Details
At minimum:

- [x] replicate `make_example_data.R` into a Python script, making sure to fix seeds where applicable
- [x] place this in a refactored `tools/artifacts` (e.g. a `datasets` sub-package)

Nice to have:

- [ ] it may not be ideal to load all data into memory, so we may want to offer a generator class or some such feature for iterating over datasets, in general
- [ ] determine if `dataCircleGSev3Sp3Train.csv` can also be cleaned up in some way
- [ ] put all included descriptions into one README for the `datasets` directory

### PR DESCRIPTION
**Pull request by Anmol-Srivastava**
### Added

- `make_example_data.py`, which contains a function that replicates what `make_example_data.R` did to generate a CSV
- `generator.py`, an un-implemented skeleton for customizable data generators

### Changed

- removed `make_example_data.R`, `make_example_data.csv`
- combined two `README.txt` files into one `README.md`
- refactored `tools/artifacts` into a `datasets` sub-package

Closes #38

### CODE REVIEW

**Comment by Anmol-Srivastava** [comment]
Will clean up formatting before merging once reviewed.

**Comment by tms-bananaquit** [comment]
Confirmed that the behavior of `make_example_batch_data` is the same as `make_example_data.R` by comparing the output:

```
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
from menelaus.datasets.make_example_data import make_example_batch_data

df = pd.read_csv('example_data.csv')
df2 = make_example_batch_data()
df2['cat'] = (df2['cat'] + 1) #R indexes from 1

df.describe()
df2.describe()

df2.drop(columns=['drift'], inplace=True)

for col in df2:
    # plt.hist(df2[col], alpha=.5)
    # plt.hist(df[col], alpha=.5)
    sns.kdeplot(df2[col], alpha=.5, label='R')
    sns.kdeplot(df[col], alpha=.5, label='py')
    plt.title(col)
    plt.legend()
    plt.show()
```

The distributions look to be close enough that it's likely the RNG differing between the two.

File: src/menelaus/datasets/generator.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# XXX   A DataGenerator may yield batch/streaming-format records, and uses
+#       user-provided settings to draw from distributions, or introduce
+#       drift - @Anmol-Srivastava
+class DataGenerator:
+    def __init__(self, mode, config):
+        self.mode = mode
+        self.config = config
```
Comment (tms-bananaquit) [inline src/menelaus/datasets/generator.py:7]: Do we want to `pass` or `raise NotImplementedError` here or something?

File: src/menelaus/datasets/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from menelaus.datasets.make_example_data import make_example_batch_data
```
Comment (tms-bananaquit) [inline src/menelaus/datasets/__init__.py:1]: I think right now `tools` is excluded from the documentation. If we're going to swap it out for a `datasets` module, then we probably want to add a sentence saying what it's for as a docstring here.

File: src/menelaus/datasets/generator.py (hunk 1 above)
Comment (tms-bananaquit) [reply src/menelaus/datasets/generator.py:7]: Also, do we need to add some dinky unit tests that just run the new functions?

**Review by tms-bananaquit — APPROVED** [review]
Looks good, assuming the tests pass.

**Pull request merged**